frame/evm/precompile/dispatch/src/lib.rs

```diff
@@ -88,8 +88,11 @@ where
 				let cost = T::GasWeightMapping::weight_to_gas(
 					post_info.actual_weight.unwrap_or(info.weight),
 				);
-
 				handle.record_cost(cost)?;
+				handle.record_external_cost(
```

Comment:
Recording the actual weight after the call execution is not enough, we must verify before dispathing the call that there is enough free room on each metric

Reply:
Done, added refund external cost hook to precompile handle